src/HeadsUpPokerEscrow.sol

```diff
@@ -851,4 +907,596 @@ contract HeadsUpPokerEscrow is ReentrancyGuard, HeadsUpPokerEIP712 {
 
         _rewardWinner(channelId, winner, wonAmount);
     }
+
+    // ------------------------------------------------------------------
+    // Force Reveal Functions
+    // ------------------------------------------------------------------
+
+    /// @notice Request force reveal of Player A's hole cards
+    /// @dev No prerequisites. Helper = B.
+    /// @param channelId The channel identifier
+    /// @param bDeck Array of 52 G1 encrypted card points (Y values), each 64 bytes
+    function requestHoleA(
+        uint256 channelId,
+        bytes[] calldata bDeck
```

Comment:
Let's store `bDeck` in the `channel[channelId]` storage. You can set it up in the `startGame` call. You won't need to store deck hash since it can be recalculated. `bDeck` of both players should be identical for the game to start. And I don't like the `bDeck` name. Let's make it simply a `deck`.